### ISSUE TITLE
Reconnect after successful connect

### ISSUE THREAD

**Issue by christianpaetz** [sushichop/cordova-plugin-wifi-manager#31]
Hi, the idea of this plugin is  - among others - to setup IOT devices. Connect to 'private' SSID,set WIFI and other parameters and then connect back to normal internet
Here is what I do:
After startup app connected to internet
window.wifiManager.connect('ABC', 'XXX, () => { ...}
do whatever is needed
window.wifiManager.disconnect('ABC', () => { ...}
(1) Now I would assume that the app connects back to Internet using 3G or WiFI. However, connection remains disconnected
(2) When I to window.wifiManager.connect('myoldWIFI', 'XXX, () => { ...} it does not work either and all Internet connections fail. Beside this the dialog report "temporary Internet" which is certainly wrong

Which of the two ways is the correct one? How can the app get back to normal Internet after disconnect from temporary WIFI ?

Thanks a lot for any hint on that.

Chris

**Comment by sushichop** [comment]
I'm not sure, but I'm guessing it's something like (2) with your device.

When you use this plugin for Android 10 (API level 29) or later, you can connect to the Internet __only through your own (cordova) app__. And this is due to the specification of new Android API, [WifiNetworkSpecifier](https://developer.android.com/reference/android/net/wifi/WifiNetworkSpecifier).

> This specifier can be used to request a local-only connection on devices that support concurrent connections (indicated via WifiManager#isStaConcurrencyForLocalOnlyConnectionsSupported() and if the initiating app targets SDK ≥ Build.VERSION_CODES.S or is a system app. These local-only connections may be brought up as a secondary concurrent connection (primary connection will be used for networks with internet connectivity available to the user and all apps).

IMO,  I think the connection depends on your device specification.
In short words, if your device supports concurrent connections, the connection will automatically connect to the Internet without disconnection(you don't need disconnect manually), probably. 

I hope this information is useful for you.

**Comment by christianpaetz** [comment]
Well, if this is true, then its impossible to connect back to Internet once the "new device" function was used. Makes the plugin essentially unusable if Internet connection is needed for the app.

**Comment by sushichop** [comment]
@christianpaetz 

I'm not sure what you are trying to do, maybe...😅

This plugin provides the feature to connect/disconnect to Wi-Fi and the limitation depends on Android API feature as described above. If your environment can connect to the Internet via W-iFi, you can make the device connect to the Internet using this plugin. See #19.

I assume what you need is the feature to check if your device can connect to the internet via Wi-Fi when you want to reconnect...?

Thanks.

**Comment by christianpaetz** [comment]
Cordova app loads Internet content and wants to pair a new IOT device. For this it must connect to local WIFI Splot of IOT device so setup home WIFISSID plus WPA. After this the IOT devices WIFI disappears ( since it now connects to the home WIFI) and the app shall connect back to the normal internet connection. The last step is what fails. After the connection to the IOT device WIFI the app remains offline. I don't have any explicit way to reconnect after the disconnect command and it does not automatically reconnect to the internet.  I think this is a quite common scenario for the use of this plugin. By the way, there is another common problem. The Cordova should use https as protocol while the IOT devices - for obvious reasons- will only communicate HTTP when waiting for the initial home WIFI/WPA. IOS blocks such request from JS level so the communication via HTTP must be done on the native part. Any good suggestion for this ?

**Comment by sushichop** [comment]
Have you tried this? #19 

In addition, unfortunately, as far as I know, the new Android API does not allow the developer to control the connection to Wi-Fi as freely as the old Android API. On the other hand,  the developer could control it freely before. `cordova-plugin-wifi-manager` is also affected by this. 

Or, if your device can work with old API, you may want to try the following. See README.
```
cordova run android --device -- --gradleArg=-PcdvTargetSdkVersion=28
```

As for HTTP, I think it is related to iOS's ATS. It is not related to `cordova-plugin-wifi-manager` and you need to read `cordova-ios` documentation.

**Comment by Diego-Varela-Silva** [comment]
I believe that problem it was that the app was stuck with a default connection that does not exists anymore.

I created this #32 to try to solve this problem

**Comment by sushichop** [comment]
@Diego-Varela-Silva Thanks for your pull request(#32). Good catch!
I have released [cordova-plugin-wifi-manager@0.4.2](https://github.com/sushichop/cordova-plugin-wifi-manager/releases/tag/0.4.2) that contains this fix.

@christianpaetz Can you try the new version? 

```
cordova plugin remove cordova-plugin-wifi-manager --save
cordova plugin add cordova-plugin-wifi-manager@0.4.2 --save
cordova plugin ls
cordova prepare
cordova run android --device
```

Thanks.

**Comment by christianpaetz** [comment]
Yes I did. I just call the disconnect method and I do see that the mobile phone is reconnecting to the original WIFi network. I even receive a window.addEventListener('online',{})  event but the app itself it not connecting back to Internet, means all connections to internet resources fail until there is a reload of the app. The current workaround is to explicitly call window.wifiManager.connect(..) with the 'old' network credentials. This works but there user experience is strange since this method shows the same message indicating a temporary  network which is not the case.

**Comment by sushichop** [comment]
@christianpaetz 
Thanks for your checking this!

> indicating a temporary network

Unfortunately, as you say, it seems the status is still indicating a temporary network...

Probably, I assume that the plugin needs the feature to bind to the network besides of Wi-Fi connection.
I don't have enough spare time to investigate this deeply for now, but I'll add this feature in the future.

Thanks for your feedback.

**Comment by christianpaetz** [comment]
Sorry, my answer was wrong. 0.42 works perfectly under Android.  Just one suggestion. To support open WIFI splots please apply this diff:
SCPWiFiManagerPlugin.m
28,29c28,33
<         NEHotspotConfiguration *configuration = [[NEHotspotConfiguration alloc] initWithSSID:ssid passphrase:passphrase isWEP:NO];
<
---
>         NEHotspotConfiguration *configuration;
>         if (passphrase.length > 7)
>         	 configuration = [[NEHotspotConfiguration alloc] initWithSSID:ssid passphrase:passphrase isWEP:NO];
>         else
>         	 configuration = [[NEHotspotConfiguration alloc] initWithSSID:ssid];

**Comment by sushichop** [comment]
@christianpaetz
Thanks for your feedback.
I have released [cordova-plugin-wifimanager@0.5.0](https://github.com/sushichop/cordova-plugin-wifi-manager/releases/tag/0.5.0) that supports open Wi-Fi network.
Now you can connect to an open Wi-Fi network by specifying empty passphrase.

`cordova-plugin-wifimanager` does not use WEP on iOS.
See [this](https://github.com/sushichop/cordova-plugin-wifi-manager/blob/main/src/ios/SCPWiFiManagerPlugin.m#L31-L33) for details.

Thanks.
